src/js/background.js

```diff
@@ -169,22 +170,45 @@ Badger.prototype = {
   updateTabList: function(){
     // Initialize the tabData/frames object if it is falsey
     this.tabData = this.tabData || {};
-    var self = this;
-    chrome.tabs.query({currentWindow: true, status: 'complete'}, function(tabs){
-      for(var i = 0; i < tabs.length; i++){
-        var tab = tabs[i];
-        self.tabData[tab.id] = {
-          frames: {
-            0: {
-              parent: -1,
-              url: tab.url
-            }
-          },
-          origins: {},
-          blockedCount: 0
-        };
+    let self = this;
+    chrome.tabs.query({}, tabs => {
+      tabs.forEach(tab => {
+        self.recordFrame(tab.id, 0, -1, tab.url);
+      });
+    });
+  },
+
+  /**
+   * Generate representation in internal data structure for frame
+   *
+   * @param tabId ID of the tab
+   * @param frameId ID of the frame
+   * @param parentFrameId ID of the parent frame
+   * @param frameUrl The url of the frame
+   */
+  recordFrame: function(tabId, frameId, parentFrameId, frameUrl) {
+    let self = this;
+    if (!self.tabData.hasOwnProperty(tabId)){
+      self.tabData[tabId] = {
+        frames: {},
+        origins: {},
+        blockedCount: 0
+      };
+    }
+    // check if this is a prerendered (bg) tab or not
+    chrome.tabs.get(tabId, function(/*tab*/){
+      if (chrome.runtime.lastError){
+        // chrome will throw error for the prerendered tabs
+        self.tabData[tabId].bgTab = true;
+      }else{
+        self.tabData[tabId].bgTab = false;
```

Comment:
This `bgTab` stuff was removed in #1480 while this PR was pending. It should be removed here.

Reply:
However this now depends on changes in #1618 so I'll wait for that to be merged before removing here.

Reply:
resolved